charts/posit-chronicle/README.md.gotmpl

```diff
@@ -0,0 +1,131 @@
+{{ template "rstudio.header" . }}
+
+{{ template "chart.versionBadge" . }}{{ template "chart.typeBadge" . }}{{ template "chart.appVersionBadge" . }}
+
+{{ template "rstudio.description" . }}
+
+{{ template "rstudio.disclaimer" . }}
+
+{{ template "rstudio.install" . }}
+
+## Usage
+
+The Chronicle chart is meant to be used in tandem with other Workbench and Connect instances.
+To enable the Chronicle agent, additional values will have to be passed to your
+Workbench and Connect values.
+
+Here's some example agent helm values to run the agent sidecar in Workbench:
+
+```yaml
+pod:
+  ...
+  sidecar:
+    - name: chronicle-agent
+      image: posit-chronicle:latest
+      volumeMounts:
+      - name: CHRONICLE_PRODUCT_CLUSTER_ID
+        value: "posit-cluster-1"
+      - name: logs
+        mountPath: "/var/lib/rstudio-server/audit"
+      env:
+      - name: CHRONICLE_SERVER_ADDRESS
+        value: "http://chronicle-server.default.svc.cluster.local"
+...
+```
+
+And here's some example agent helm values for Connect, where we utilize an API key stored as a k8s secret
+to scrape the Connect metrics REST service:
+
+```yaml
+pod:
+  ...
+  sidecar:
+    - name: chronicle-agent
+      image: posit-chronicle:latest
+      env:
+      - name: CHRONICLE_PRODUCT_CLUSTER_ID
+        value: "posit-cluster-1"
+      - name: CHRONICLE_SERVER_ADDRESS
+        value: "http://chronicle-server.default.svc.cluster.local"
+      - name: CONNECT_API_KEY
+        valueFrom:
+          secretKeyRef:
+            name: connect
+            key: apikey
+```
+
+Note that it will be up to the user to provision that Connect API key Kubernetes secret.
+
+## Storage Configuration
+
+Chronicle can be configured to store its data in a local kubernetes volime, in S3,
+or in both.
+
+The default configuration uses a local volume, which is suitable if you'd like to 
+access and analyze the data within your cluster:
+
+``` yaml
+config:
+  localStorage:
+    enabled: true
+    location: "./chronicle-data"
+    retentionPeriod: "30d"
+```
+
+`RetentionPeriod` accepts a duration string input. `0` implies infinite retention, disabling file expiration.
+For example:
+`1s` for 1 second, `5m` for 5 minutes, `12h` for 12 hours, `7d` for one week, `365d` for one year, `0` for unbound retention.
+Units shorter than seconds or longer than days, such as milliseconds and weeks, are not supported.
+
+
+You can disable local storage by setting `localStorage.enabled` to `false`, and you can enable S3
+storage by setting `S3Storage.enabled` to `true`. Enabling both is also acceptable,
+and the server will store to both places. When using S3, you must also set the `S3Storage.Bucket`
+parameter, like so:
```

Comment:
```suggestion
`retentionPeriod` controls how long usage data are kept. For example, `"120m"`
for 120 minutes, `"36h"` for 36 hours, `14d` for two weeks, or `"0"` for
unbounded retention. (Units smaller than seconds or larger than days are not
supported.)

You can also persist data to AWS S3 instead of (or in addition to) local
storage:
```